core/src/main/java/org/infinispan/container/impl/EntryFactoryImpl.java

```diff
@@ -197,17 +201,25 @@ private CacheEntry getFromContext(InvocationContext ctx, Object key) {
       return cacheEntry;
    }
 
-   private CacheEntry getFromContainer(Object key, int segment, boolean isOwner, boolean writeOperation) {
+   private boolean isPrimaryOwner(boolean isOwner, int segment) {
+      if (distributionManager != null) {
+         return distributionManager.getCacheTopology().getSegmentDistribution(segment).isPrimary();
+      } else {
+         return isOwner;
```

Comment:
Actually `distributionManager == null` only in `LOCAL` caches, so you can return `true` here and remove the parameter.